bid-distribution/src/settlement_claims.rs

```diff
@@ -0,0 +1,168 @@
+                .sum();
+
+            let stake_sam_percentage = sam_target_stake / (sam_target_stake + mnde_target_stake);
+            let initial_sam_stake = (Decimal::from(total_active_stake) * stake_sam_percentage)
+                .to_u64()
+                .unwrap();
+            let effective_sam_stake: u64 =
+                min(initial_sam_stake, max_wanted_stake.to_u64().unwrap());
+            let effective_total_claim = Decimal::from(effective_sam_stake) * effective_bid;
+            let marinade_fee_claim = (effective_total_claim * marinade_payment_percentage)
+                .to_u64()
+                .unwrap();
+            let stakers_total_claim = (effective_total_claim
+                * (Decimal::from(1) - marinade_payment_percentage))
+                .to_u64()
+                .unwrap();
+
+            let mut claims = vec![];
+            let mut claims_amount = 0;
+
+            for ((withdraw_authority, stake_authority), stake_metas) in grouped_stake_metas {
+                if !stake_authority_filter(stake_authority) {
+                    continue;
+                }
+                let stake_accounts: HashMap<_, _> = stake_metas
+                    .iter()
+                    .map(|s| (s.pubkey, s.active_delegation_lamports))
+                    .collect();
+                let active_stake: u64 = stake_accounts.values().sum();
+                if active_stake > 0 {
+                    let claim_amount = ((Decimal::from(active_stake)
+                        / Decimal::from(total_active_stake))
+                        * Decimal::from(stakers_total_claim))
+                    .to_u64()
+                    .unwrap();
+                    if claim_amount > 0 {
+                        claims.push(SettlementClaim {
+                            withdraw_authority: **withdraw_authority,
+                            stake_authority: **stake_authority,
+                            stake_accounts,
+                            claim_amount,
+                            active_stake,
+                        });
+                        claims_amount += claim_amount;
+                    }
+                }
+            }
+
+            assert!(
+                claims_amount <= stakers_total_claim,
+                "Claims amount is bigger than stakers total claim"
+            );
+
+            claims.push(SettlementClaim {
+                withdraw_authority: *settlement_config.marinade_withdraw_authority(),
+                stake_authority: *settlement_config.marinade_stake_authority(),
+                stake_accounts: marinade_fee_deposit_stake_accounts.clone(),
+                claim_amount: marinade_fee_claim,
+                active_stake: marinade_fee_deposit_stake_accounts.values().sum(),
+            });
+            claims_amount += marinade_fee_claim;
+
+            assert!(
+                claims_amount <= stakers_total_claim + marinade_fee_claim,
```

Comment:
Should not be here rather considered the `effective_total_claim` that should be more or less (that's and issue that because of calculation it is not equal probably?) equal to sum of `stakers_total_claim + marinade_fee_claim`?

Reply:
I wanted to double check that the sum of all claims doesn't exceed the effective_total_claim, which is indeed (stakers_total_claim + marinade_fee_claim)